backend/src/main/java/harustudy/backend/auth/infrastructure/CompositeOauthClient.java

```diff
@@ -0,0 +1,39 @@
+package harustudy.backend.auth.infrastructure;
+
+import harustudy.backend.auth.dto.OauthTokenResponse;
+import harustudy.backend.auth.exception.InvalidProviderNameException;
+import lombok.RequiredArgsConstructor;
+
+import java.util.List;
+import java.util.Map;
+
+@RequiredArgsConstructor
+public class CompositeOauthClient implements OauthClient {
```

Comment:
```suggestion
@RequiredArgsConstructor
@Component(value = "oauthClient")
public class CompositeOauthClient implements OauthClient {
```

맨 처음 보았을 때 왜 이 클래스는 Bean으로 등록하지 않지? 라는 생각이 들었는데 다른 Config 파일에서 등록해주고 있었군요!
어차피 이름을 통해 CompositeOauthClient를 찾는 것이 불가피하다면 설정 파일을 분리하는 방향보다 이 클래스에서 `@Component(value = ?)`로 바로 이름을 지정하여 등록해서 변경 포인트를 줄이는 방향은 어떻게 생각하시나요?

만약 설정 파일을 유지하게 된다면 그것대로 또 설정 파일의 위치를 어떤 패키지에 두어야 할 지 다시 고민하게 될 것 같습니다!

Reply:
좋은 생각입니다!

저도 configuration 파일이나 빈에대한 고민이 있었는데 덕분에 훨씬 간단해졌습니다 👍